client/asset/btc/spv_wrapper.go

```diff
@@ -1030,36 +1032,50 @@ func (w *spvWallet) walletUnlock(pw []byte) error {
 	return w.Unlock(pw)
 }
 
-func (w *spvWallet) getBlockHeader(blockHash *chainhash.Hash) (*blockHeader, error) {
+// getBlockHeader gets the *blockHeader for the specified block hash. It also
+// returns a bool value to indicate whether this block is a part of main chain.
+// For orphaned blocks header.Confirmations is negative (typically -1).
+func (w *spvWallet) getBlockHeader(blockHash *chainhash.Hash) (header *blockHeader, mainchain bool, err error) {
 	hdr, err := w.cl.GetBlockHeader(blockHash)
 	if err != nil {
-		return nil, err
+		return nil, false, err
 	}
 
 	tip, err := w.cl.BestBlock()
 	if err != nil {
-		return nil, fmt.Errorf("BestBlock error: %v", err)
+		return nil, false, fmt.Errorf("BestBlock error: %v", err)
 	}
 
 	blockHeight, err := w.cl.GetBlockHeight(blockHash)
 	if err != nil {
-		return nil, err
+		return nil, false, err
+	}
+
+	confirmations := int64(-1)
+	mainchain = w.blockIsMainchain(blockHash, blockHeight)
+	if mainchain {
+		confirmations = int64(confirms(blockHeight, tip.Height))
+	}
+	if tip.Height < blockHeight { // if tip is less, may be rolling back, so just mock dcrd/dcrwallet
+		confirmations = 0
 	}
```

Comment:
Even if it's not mainchain? If it's mainchain and `tip.Height < blockHeight`, `confirmations` would be set to 0 by the `confirms` fn.

Reply:
I've re-checked with https://github.com/decred/dcrdex/pull/2219 once again,  there is slight discrepancy which should be addressed by https://github.com/decred/dcrdex/pull/2225/commits/3ded90408babb40df8969e0bc893c85e2add313c, thanks for pointing it out!

> If it's mainchain and tip.Height < blockHeight, confirmations would be set to 0 by the confirms fn.

I believe when we have `tip.Height < blockHeight` we can't also have `mainchain=true`, right ? So this shouldn't be an issue. But to be absolutely sure I've changed this condition a bit to only trigger on `mainchain=false`.

Reply:
> I've changed this condition a bit to only trigger on `mainchain=false`.

I think @itswisdomagain's point was that this `tip.Height < blockHeight` check is not required at all because the `confirms` func handles this, unlike with dcr where we don't have that helper.  If `blockIsMainchain` says `false`, I don't believe we would overrule that.

In other words, with no knowledge about `confirms` functionality in this `<` situation, the code would be:

```go
	if mainchain {
		if tip.Height < blockHeight { // if tip is less, may be rolling back, so just mock dcrd/dcrwallet
			confirmations = 0
		} else {	
			confirmations = int64(confirms(blockHeight, tip.Height))
		}
	}

	return &blockHeader{
```

But `confirms` handles that already, so it's just

```go
	if mainchain {
		confirmations = int64(confirms(blockHeight, tip.Height))
	}

	return &blockHeader{
```